app/controllers/merchants/items_controller.rb

```diff
@@ -1,5 +1,32 @@
 class Merchants::ItemsController < ApplicationController
+  before_action :get_item, only: [:show, :edit]
+
   def index
     @items = Merchant.find(params[:merchant_id]).items
   end
-end
\ No newline at end of file
+
+  def update
+    get_item.update(item_params)
+    flash.notice = "Item has been updated!"
+    redirect_to params[:previously] || params[:item][:previously]
```

Comment:
I think the instructors wanted us to break these into two separate controller actions, but I could be misremembering.  I know we have the same thing in our branch so we could possibly address this as a group.

Reply:
yeah, we definitely want to get rid of that 'previously' stuff. I will work on that and push those changes before we merge.